dotnet/src/MicroFocus.CafApi.QueryBuilder.Matcher/MatcherQueryVisitor.cs

```diff
@@ -41,145 +44,358 @@ public bool GetResult()
         }
 
         public void VisitGreaterThanOrEquals(IMatcherFieldSpec<Document> fieldSpec, long value)
         {
-            throw new NotImplementedException();
+            _isMatch = GetLongStream(fieldSpec).Any(v => v >= value);
         }
 
         public void VisitGreaterThanOrEquals(IMatcherFieldSpec<Document> fieldSpec, string value)
         {
-            throw new NotImplementedException();
+            string valueToCompare = fieldSpec.IsCaseInsensitive ? value.ToUpper(new CultureInfo("en-US")) : value;
+
+            _isMatch = GetStringValues(fieldSpec).Any(v => v.CompareTo(valueToCompare) >= 0);
         }
 
         public void VisitExists(IMatcherFieldSpec<Document> fieldSpec)
         {
-            throw new NotImplementedException();
+            _isMatch = GetFieldValues(fieldSpec).Any();
         }
 
         public void VisitEmpty(IMatcherFieldSpec<Document> fieldSpec)
         {
-            throw new NotImplementedException();
+            _isMatch = !GetStringValues(fieldSpec).Any(v => v.Length > 0);
         }
 
         public void VisitOr(IEnumerable<Filter<IMatcherFieldSpec<Document>>> filters)
         {
-            throw new NotImplementedException();
+            foreach (Filter<IMatcherFieldSpec<Document>> filter in filters)
+            {
+                filter.Invoke(this);
+                if (_isMatch)
+                {
+                    break;
+                }
+            }
         }
 
         public void VisitAnd(IEnumerable<Filter<IMatcherFieldSpec<Document>>> filters)
         {
-            throw new NotImplementedException();
+            _isMatch = filters.All(filter => filter.IsMatch(_document, _allFullTextFieldSpecs));
         }
 
         public void VisitNot(Filter<IMatcherFieldSpec<Document>> filter)
         {
-            throw new NotImplementedException();
+            filter.Invoke(this);
+            _isMatch = !_isMatch;
         }
 
         public void VisitFullText(FullTextFilter fullTextFilter)
         {
-            throw new NotImplementedException();
+            VisitFieldFullTextImpl(_allFullTextFieldSpecs, fullTextFilter);
         }
 
         public void VisitFieldFullText(
             IEnumerable<IMatcherFieldSpec<Document>> fieldSpecs,
             FullTextFilter fullTextFilter
         )
         {
-            throw new NotImplementedException();
+            VisitFieldFullTextImpl(fieldSpecs, fullTextFilter);
+        }
+
+        private IEnumerable<IMatcherFieldValue> GetFieldValues(IMatcherFieldSpec<Document> fieldSpec)
+        {
+            return fieldSpec.GetFieldValues(_document);
+        }
+
+        private IEnumerable<long> GetLongStream(IMatcherFieldSpec<Document> fieldSpec)
+        {
+            return GetStringValues(fieldSpec).Where(v =>
+            {
+                return long.TryParse(v, NumberStyles.AllowLeadingSign, null, out long xi);
+            }).Select(v => long.Parse(v));
```

Comment:
With the `.TryParse()` you are passing `NumberStyles.AllowLeadingSign` which you are not on the `.Parse()`.  This could lead to a situation where the trial succeeds but the actual parse doesn't.  Really what you should do here is to use an [Anonymous Types](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/types/anonymous-types) to avoid the need for the second parse.